### PR TITLE
Color inherits from textinput

### PR DESCRIPTION
**Pull request by ppizarror**

### CODE REVIEW

**Comment by ppizarror** [comment]
@anxuae what do you think about this? You can test it in MultiInput example. I think it's bulletproof, can you test please 😄 

epic

**Comment by ppizarror** [comment]
TODO:
- HEX support
- Previsualization

**Comment by anxuae** [comment]
It seems really good! Tested and approved on my side.
With color visualization it will be complete.

May be you need also to overwrite the `get_value` method to return a tuple of int (_TYPE_RGB) or a string (_TYPE_HEX).

**Comment by ppizarror** [comment]
@anxuae Finished HEX format, `get_value` and color previsualization.
TODO: Update README, add tests

Can you test the widget please? Any non valid input, extreme cases, etc. Thanks!

epic3

File: pygameMenu/widgets/colorinput.py [hunk 1]
```diff
@@ -0,0 +1,417 @@
...
+class ColorInput(TextInput):
+    """
+    Color input widget.
+    """
+
+    def __init__(self,
+                 label='',
+                 colorinput_id='',
+                 color_type=_TYPE_RGB,
+                 input_comma=',',
```
Comment (anxuae) [inline pygameMenu/widgets/colorinput.py:52]: input_sep ?

**Comment by anxuae** [comment]
All seems good.

But callbacks `onreturn` and `onchange` does not work.

I think you missed to call `self.apply()` and `self.change()` in the `update()` method of `CloreInput` class.

**Comment by ppizarror** [comment]
> All seems good.
> 
> But callbacks `onreturn` and `onchange` does not work.
> 
> I think you missed to call `self.apply()` and `self.change()` in the `update()` method of `ColorInput` class.

That calls are made by the super method. My mistake was in how kwargs were passed to the constructor.
As ColorInput extends from TextInput, I had to unzip kwargs variable. Now `onreturn` and `onchange` works seamlessly

File: pygameMenu/widgets/colorinput.py (hunk 1 above)
Comment (ppizarror) [reply pygameMenu/widgets/colorinput.py:52]: you mean to change the name of the parameter?

Comment (anxuae) [reply pygameMenu/widgets/colorinput.py:52]: yes, because is not necessary a comma ;)
however there is no matter, it's working like that.

Comment (ppizarror) [reply pygameMenu/widgets/colorinput.py:52]: ah true, so I failed in both testing (assumed that there's a comma) and the of the parameters. I'll change this now.

**Pull request merged**